janus.c

```diff
@@ -4580,11 +4606,26 @@ gint main(int argc, char *argv[])
 	item = janus_config_get(config, config_nat, janus_config_type_item, "nat_1_1_mapping");
 	if(item && item->value) {
 		JANUS_LOG(LOG_INFO, "Using nat_1_1_mapping for public IP: %s\n", item->value);
-		if(!janus_network_string_is_valid_address(janus_network_query_options_any_ip, item->value)) {
-			JANUS_LOG(LOG_WARN, "Invalid nat_1_1_mapping address %s, disabling...\n", item->value);
-		} else {
-			nat_1_1_mapping = item->value;
-			janus_set_public_ip(nat_1_1_mapping);
+		char **list = g_strsplit(item->value, ",", -1);
+		char *index = list[0];
+		GPtrArray *ips = g_ptr_array_new_with_free_func(janus_free_string);
+		if(index != NULL) {
+			int i=0;
+			while(index != NULL) {
+				if(strlen(index) > 0) {
+					if(!janus_network_string_is_valid_address(janus_network_query_options_any_ip, index)) {
+						JANUS_LOG(LOG_WARN, "Invalid nat_1_1_mapping address %s, skipping...\n", index);
+					} else {
+						g_ptr_array_add(ips, g_strdup(index));
```

Comment:
This isn't checking if there are duplicates. It may be easier to use a `GHashTable` instead of an array in this parsing phase (where each IP is the key, and the value can be a `TRUE` turned to pointer), and then duplicate the `GList` returned by [g_hash_table_get_keys](https://developer.gnome.org/glib/stable/glib-Hash-Tables.html#g-hash-table-get-keys) and keep that instead.

As an alternative, you can use the hash table as the struct to keep those IPs in general, which may be a better option as another public IP may come from STUN (if configured) and that may trigger duplicates as well. In that case I'd simply add a new public method called something like `janus_add_public_ip()` that updates the table. Then, whenever the list of IPs is needed for the SDP, you can just get the list of keys as shown above.

Reply:
I was trying to maintain backwards compatibility on the existing methods, but having a `janus_add_public_ip` method is a good idea. I'll update the PR accordingly, thanks!